stan/math/prim/prob/wiener_full_lpdf.hpp

```diff
@@ -0,0 +1,618 @@
+    return ret_t(0);
+  }
+  const size_t N = max_size(y, a, v, w, t0, sv, sw, st0);
+  if (!N) {
+    return ret_t(0);
+  }
+  scalar_seq_view<T_y_ref> y_vec(y_ref);
+  scalar_seq_view<T_a_ref> a_vec(a_ref);
+  scalar_seq_view<T_v_ref> v_vec(v_ref);
+  scalar_seq_view<T_w_ref> w_vec(w_ref);
+  scalar_seq_view<T_t0_ref> t0_vec(t0_ref);
+  scalar_seq_view<T_sv_ref> sv_vec(sv_ref);
+  scalar_seq_view<T_sw_ref> sw_vec(sw_ref);
+  scalar_seq_view<T_st0_ref> st0_vec(st0_ref);
+  const size_t N_y_t0 = max_size(y, t0, st0);
+
+  for (size_t i = 0; i < N_y_t0; ++i) {
+    if (y_vec[i] <= t0_vec[i]) {
+      std::stringstream msg;
+      msg << ", but must be greater than nondecision time = " << t0_vec[i];
+      std::string msg_str(msg.str());
+      throw_domain_error(function_name, "Random variable", y_vec[i], " = ",
+                         msg_str.c_str());
+    }
+  }
+  size_t N_beta_sw = max_size(w, sw);
+  for (size_t i = 0; i < N_beta_sw; ++i) {
+    if (unlikely(w_vec[i] - .5 * sw_vec[i] <= 0)) {
+      [&]() STAN_COLD_PATH {
+        std::stringstream msg;
+        msg << ", but must be smaller than 2*(A-priori bias) = "
+            << 2 * w_vec[i];
+        std::string msg_str(msg.str());
+        throw_domain_error(function_name,
+                           "Inter-trial variability in A-priori bias",
+                           sw_vec[i], " = ", msg_str.c_str());
+      }();
+    }
+    if (unlikely(w_vec[i] + .5 * sw_vec[i] >= 1)) {
+      [&]() STAN_COLD_PATH {
+        std::stringstream msg;
+        msg << ", but must be smaller than 2*(1-A-priori bias) = "
+            << 2 * (1 - w_vec[i]);
+        std::string msg_str(msg.str());
+        throw_domain_error(function_name,
+                           "Inter-trial variability in A-priori bias",
+                           sw_vec[i], " = ", msg_str.c_str());
+      }();
+    }
+  }
+
+  const T_partials_return log_error_density
+      = log(1e-6);                                 // precision for density
+  const auto error_bound = precision_derivatives;  // precision for
+  // derivatives (controllable by user)
+  const auto log_error_derivative = log(error_bound);
+  const T_partials_return absolute_error_hcubature = 0.0;
+  const T_partials_return relative_error_hcubature
+      = .9 * error_bound;  // eps_rel(Integration)
+  const T_partials_return log_error_absolute = log(1e-12);
+  const int maximal_evaluations_hcubature = 6000;
+  T_partials_return log_density = 0.0;
+  auto ops_partials = make_partials_propagator(y_ref, a_ref, t0_ref, w_ref,
+                                               v_ref, sv_ref, sw_ref, st0_ref);
+  ret_t result = 0;
+
+  // calculate density and partials
+  for (size_t i = 0; i < N; i++) {
+    if (sw_vec[i] == 0 && st0_vec[i] == 0) {
+      result += wiener_lpdf<propto>(y_vec[i], a_vec[i], t0_vec[i], w_vec[i],
+                                    v_vec[i], sv_vec[i], precision_derivatives);
+      continue;
+    }
```

Comment:
So you mean the user now can call the diffusion model in the .stan file with the following alternatives:

The seven-parameter model: 
`target += wiener_lpdf(y | a, t0, w, v, sv, sw, st0);`

The five parameter model: 
`target += wiener_lpdf(y | a, t0, w, v, sv);`

Also the four parameter model?
`target += wiener_lpdf(y | a, t0, w, v);`

Is this possible? Or do we need to give `sv` in the five parameter implementation.

If this is possible, we can delete the above highlighted lines. But we should then give examples in the documentation for the different possibilities to call the function.

Is it then still possible to compute the four parameter model with the following line:
`target += wiener_lpdf(y | a, t0, w, v, 0, 0, 0);`

Reply:
> Is this possible?

Yes. Since they each have different numbers of arguments the compiler will be able to deduce which one to call

> Is it then still possible to compute the four parameter model with the following line:
`target += wiener_lpdf(y | a, t0, w, v, 0, 0, 0);`

No that would call the wiener with 5 arguments. If they are the same mathematically with the extra terms set to zero then you would get the four parameter model, but otherwise users should be calling the 4 parameter model directly

Reply:
We should leave the mentioned lines in the code. 

For the case that users do use the function call like this:
`target += wiener_lpdf(y | a, t0, w, v, 0, 0, 0);`,
as this is the way I described in the paper how to call the 4- and 5-parameter model. In this case we need the forwarding to wiener5 in wiener_full. Additionally, we can describe in the documentation that the preferred function call is without the zeros at the end. 

I added tests to call the function without the zeros. This works:
`wiener_lpdf(y, a, t0, w, v, sv);`

Do you intend to delete the old version of the `wiener` implementation? The one that already exists in Stan? Else the 4-parameter-call would lead to the old version:
`wiener_lpdf(y, a, t0, w, v);`
But I would like to call my implementation of the 4-parameter diffusion model. For this, I added in `wiener5_lpdf` the following lines:
```
template <bool propto = false, typename T_y, typename T_a, typename T_t0,
          typename T_w, typename T_v>
inline auto wiener_lpdf(const T_y& y, const T_a& a, const T_t0& t0,
                        const T_w& w, const T_v& v,
                        const double& precision_derivatives = 1e-4) {
	return wiener_lpdf(y, a, t0, w, v, 0, precision_derivatives);
}  // end wiener_lpdf
```